ui/src/validation.ts

```diff
@@ -0,0 +1,62 @@
+import validatejs from "validate.js"
+import { derived, writable, Readable, get } from "svelte/store"
+
+export enum ValidationStatus {
+  NotStarted = "NOT_STARTED",
+  FormatValidation = "FORMAT_VALIDATION",
+  Pending = "PENDING",
+  Error = "ERROR",
+  Success = "SUCCESS"
+}
+
+type ValidationState =
+  { status: ValidationStatus.NotStarted } |
+  { status: ValidationStatus.FormatValidation; input: string } |
+  { status: ValidationStatus.Pending; input: string } |
+  { status: ValidationStatus.Error; input: string; message: string } |
+  { status: ValidationStatus.Success; input: string }
+
+interface ValidationStore extends Readable<ValidationState> {
+  validate: (input: string) => void;
+}
+
+const VALID_NAME_MATCH = new RegExp("^[a-z0-9][a-z0-9_-]+$", "i");
+const constraints = {
```

Comment:
These seem to be specific to orgs, should they live here?

Reply:
I'm planning on this sort of thing being a context-specific variable that consumers can add